frontend/public/components/catalog-item-details.jsx

```diff
@@ -0,0 +1,107 @@
+import * as _ from 'lodash-es';
+import * as PropTypes from 'prop-types';
+import {Link} from 'react-router-dom';
+import {Modal} from 'patternfly-react/dist/esm/components/Modal';
+import {CatalogItemHeader} from 'patternfly-react-extensions/dist/esm/components/CatalogItemHeader';
+import {PropertiesSidePanel, PropertyItem} from 'patternfly-react-extensions/dist/esm/components/PropertiesSidePanel';
+
+import {normalizeIconClass} from './catalog-item-icon';
+import {ClusterServicePlanModel} from '../models';
+import {k8sGet} from '../module/k8s';
+import {Timestamp} from './utils';
+
+export class CatalogTileDetails extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      plans: [],
+    };
+  }
+
+  componentDidMount() {
+    const { obj, kind } = this.props.item;
+    if (kind === 'ClusterServiceClass') {
+      this.getPlans(obj);
+    }
+  }
+
+  getPlans(obj) {
+    k8sGet(ClusterServicePlanModel, null, null, {queryParams: { fieldSelector: `spec.clusterServiceClassRef.name=${obj.metadata.name}`} })
+      .then(({items: plans}) => {
+        this.setState({
+          plans: _.orderBy(plans, ['spec.externalMetadata.displayName', 'metadata.name'])
+        });
+      });
+  }
+
+  render () {
+    const { closeOverlay } = this.props;
+    const { obj,
+      tileName,
+      tileImgUrl,
+      tileIconClass,
+      tileProvider,
+      tileDescription,
+      href,
+      supportUrl,
+      longDescription,
+      documentationUrl,
+      sampleRepo } = this.props.item;
+    const { plans } = this.state;
+
+    const vendor = tileProvider ? `Provided by ${tileProvider}` : null;
+    const iconClass = tileIconClass ? normalizeIconClass(tileIconClass) : null;
+    const creationTimestamp = _.get(obj, 'metadata.creationTimestamp');
+
+    const supportUrlLink = <a href={supportUrl} target="_blank" className="co-external-link" rel="noopener noreferrer">Get Support</a>;
+    const documentationUrlLink = <a href={documentationUrl} target="_blank" className="co-external-link" rel="noopener noreferrer">{documentationUrl}</a>;
+    const sampleRepoLink = <a href={sampleRepo} target="_blank" className="co-external-link" rel="noopener noreferrer">{sampleRepo}</a>;
+    const planItems = _.map(plans, plan => <li key={plan.metadata.uid}>{plan.spec.description || plan.spec.externalName}</li>);
+
+    return (
+      <React.Fragment>
+        <Modal.Header>
+          <Modal.CloseButton onClick={closeOverlay} />
+          <CatalogItemHeader
+            title={tileName}
+            vendor={vendor}
+            iconClass={iconClass}
+            iconImg={tileImgUrl} />
+        </Modal.Header>
+        <Modal.Body>
+          <div className="co-catalog-page__overlay-body">
+            <PropertiesSidePanel>
+              <Link className="btn btn-primary co-catalog-page__overlay-create" to={href} role="button">{`Create ${href.includes('source-to-image') ? 'Application' : 'Service Instance'}`}</Link>
```

Comment:
```suggestion
              <Link className="btn btn-primary co-catalog-page__overlay-create" to={href} role="button">Create {obj.kind === 'ImageStream' ? 'Application' : 'Service Instance'}</Link>
```

Although it might be better to set a create label property on the item when it's created.